### DISCUSSION TITLE
[コラム] Node.jsのエラーファーストコールバックのimportについて

### DISCUSSION THREAD

**Discussion by nagayou** [asciidwango/js-primer#1705] [質問] [answered]
## 該当ページ

- https://jsprimer.net/use-case/nodecli/read-file/#node-error-first-callbak

## 説明

```
import * as fs from "node:fs/promises";
```
とあるのですが、この場合は
```
import * as fs from "node:fs";
```
というふうにpromisesは記述しないのではないでしょうか？

**Comment by azu** [comment, marked as answer]
`node:fs/promises` はPromise版のAPIが含まれている一方で、コールバック版のAPIは含まれていないものになっています。

- [Promises API](https://nodejs.org/api/fs.html#promises-api)

そのため、コールバック関数を受け取るバージョンのAPIを使う場合は、`import * as fs from "node:fs";`とする形ですね。

```js
import * as fs from "node:fs";
fs.readFile(filePath, (error, result) => { 
  // コールバック版
})
```

**Reply by nagayou** [reply]
ご回答ありがとうございます。
私が、文章中の「fs/promiseモジュール」と「fsモジュール」を区別できていませんでした。
コードブロック内の`// fsモジュールにはエラーファーストコールバックを扱う非同期APIも含まれる`は、その下の`import * as fs from "node:fs/promises";`の説明ではなく、「今回はfs/promiseモジュールだけど、fsモジュールの場合はコールバックを扱うAPIがある」という注釈なのですね。
初学者の私にとって、少し読解力を要する箇所でした。
